lib/bedrock-item.js

```diff
@@ -0,0 +1,207 @@
+          can_place_on: this.canPlaceOn,
+          can_destroy: this.canDestroy
+        }
+      }
+    }
+
+    /** @returns {BedrockItem} */
+    static fromNetwork (item, serverAuthoritative = true) {
+      return new BedrockItem(
+        item.network_id,
+        item.count,
+        item.metadata,
+        item.extra.nbt?.nbt ?? null,
+        item.extra.can_place_on,
+        item.extra.can_destroy,
+        item.stack_id
+      )
+    }
+
+    get customName () {
+      if (Object.keys(this).length === 0) return null
+      return this.nbt?.value?.display?.value?.Name?.value ?? null
+    }
+
+    set customName (newName) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      if (!this.nbt.value.display) this.nbt.value.display = { type: 'compound', value: {} }
+      this.nbt.value.display.value.Name = { type: 'string', value: newName }
+    }
+
+    get customLore () {
+      if (Object.keys(this).length === 0) return null
+      return this.nbt?.value?.display?.value?.Lore?.value.value ?? null
+    }
+
+    set customLore (newLore) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      if (!this.nbt.value.display) this.nbt.value.display = { type: 'compound', value: {} }
+      this.nbt.value.display.value.Lore = { type: 'list', value: { type: 'string', value: newLore } }
+    }
+
+    get repairCost () {
+      if (Object.keys(this).length === 0) return 0
+      return this.nbt?.value.RepairCost?.value ?? 0
+    }
+
+    set repairCost (value) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      this.nbt.value.RepairCost = { type: 'int', value }
+    }
+
+    get enchants () {
+      if (Object.keys(this).length === 0) return 0
+      if (!this.nbt?.value?.ench) return []
+      return this.nbt.value.ench.value.value.map((ench) => ({
+        lvl: ench.lvl.value,
+        name: registry.enchantments[ench.id.value]?.name || null // TODO: bedrock enchantments
+      }))
+    }
+
+    set enchants (normalizedEnchArray) {
+      const enchs = normalizedEnchArray.map(({ name, lvl }) => ({
+        id: { type: 'short', value: registry.enchantmentsByName[name].id },
+        lvl: { type: 'short', value: lvl }
+      }))
+
+      if (enchs.length !== 0) {
+        if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+        this.nbt.value.ench = { type: 'list', value: { type: 'compound', value: enchs } }
+      }
+    }
+
+    get durabilityUsed () {
+      if (Object.keys(this).length === 0) return null
+      return this.nbt?.value?.Damage?.value ?? 0
+    }
+
+    set durabilityUsed (value) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      this.nbt.value.Damage = { type: 'int', value }
+    }
+
+    get spawnEggMobName () {
+      return this.name.replace('_spawn_egg', '')
+    }
+
+    /** @returns {boolean} */
+    canBePlacedOnBlock (block) {
```

Comment:
canBePlacedOnBlock -> canPlaceOn

Reply:
canPlaceOnBlock? the field with the array is called canPlaceOn

Reply:
Then those fields need to be private or be called something else

Reply:
if private then I should add methods to add/remove blocks it can be placed on/destroy
or would something like canPlaceOnArray and canDestroyArray work? I can't think of a better name

Reply:
If these is these fields are exposed, this would need to be added to the pc version also (`canPlaceOn()`, `canDestoryBlock()`). This includes the stackId field. There shouldn't need to be version checks on basic prismarine-item functions (except for the protocol import/export functions) as it's meant to abstract the item, not just re-expose protocol specific internals. 

In the future, they might add these functions to the NBT. So for now adding `breakableBlocks` and `placeableBlocks` field to prismarine-item may make sense, and on bedrock these things refer to the special fields sent over the network.